yoti-sdk-api/src/main/java/com/yoti/api/client/shareurl/policy/DynamicPolicy.java

```diff
@@ -10,34 +10,40 @@
 
 import com.yoti.api.attributes.AttributeConstants;
 import com.yoti.api.client.shareurl.constraint.Constraint;
+import com.yoti.api.client.shareurl.policy.profile.IdentityProfile;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 /**
  * Set of data required to request a sharing transaction
  */
-public class DynamicPolicy {
+public final class DynamicPolicy {
 
-    @JsonProperty("wanted")
+    @JsonProperty(Property.WANTED)
     private final Collection<WantedAttribute> wantedAttributes;
 
-    @JsonProperty("wanted_auth_types")
+    @JsonProperty(Property.WANTED_AUTH_TYPES)
     private final Set<Integer> wantedAuthTypes;
 
-    @JsonProperty("wanted_remember_me")
+    @JsonProperty(Property.WANTED_REMEMBER_ME)
     private final boolean wantedRememberMe;
 
-    @JsonProperty("wanted_remember_me_optional")
+    @JsonProperty(Property.WANTED_REMEMBER_ME_OPTIONAL)
     private final boolean wantedRememberMeOptional;
 
+    @JsonProperty(Property.IDENTITY_PROFILE_REQUIREMENTS)
+    private final IdentityProfile identityProfile;
```

Comment:
If the wantedRememberMe field and constructor param are best with "Optional" suffix - should we perhaps similarly mark the new identityProfile as such with "identityProfileOptional"?

Reply:
@BenSmithYoti I _think_ this is in fact saying that "the wanted remember me is optional in the share", rather than _itself_ being an optional property which may or not be provided for the policy to be built